### DISCUSSION TITLE
Help with Go-Lang Templates

### DISCUSSION THREAD

**Discussion by regice202** [JanDeDobbeleer/oh-my-posh#1165] [Troubleshoot] [answered]
Hello! I just recently started using oh my posh and I'm trying to configure it to my liking and also fix a weird bug I'm having with Windows Terminal. The issue is that in terminal, for some reason, even without modifying the unicorn theme file, I get this weird gap in the PS console. ![WindowsTerminal_YnHgiBaCYi](https://user-images.githubusercontent.com/8492766/140266673-9c355c72-f66e-479b-801a-853ec97a6888.png) So I'm trying to use templates to evaluate whether or not the host process is Windows Terminal (which I've done with the PS code I'm using in the template), then remove the character if so; otherwise insert it to fix the weird graphical glitch associated with the theme normally. Here's the template, any help is much appreciated! `"postfix_template": "{{with $WTS := (GWMI Win32_Process | ? ProcessId -eq .PID).ParentProcessId}}{{if contains \"WindowsTerminal\" (Get-Process -Id $WTS).Name}}\u2800{{end}}"`

**Comment by JanDeDobbeleer** [comment, marked as answer]
@regice202 you can't evaluate PowerShell code in the templates. What you can do is access environment variables so the code you have could be in your `$PROFILE` and set an environment variable's value. That way, you can always call it using `{{ .Env.WINTERM eq "1" }}` for example and set the correct icon. There's also the possibility to fix the space by [using a special character](https://ohmyposh.dev/docs/faq#windows-terminal-unexpected-space-between-segmentstext).

**Reply by regice202** [reply]
Hey @JanDeDobbeleer, thanks for the quick reply! So I have the env var setup now and it's working properly, but for some reason it's still not working... `"postfix_template": "{{if .Env.WindowsTerminalOpen eq \"1\"}}\u2800{{end}}"` Thanks again for the help and awesome work!

Edit: Just realized there isn't such thing as a `"postfix_template":`, at least not specified in the docs of the website... Also that character you mentioned is the one causing problems in my other normal PS sessions, drawing the weird unknown character box. So any chance you could make all the tags able to have templates?... That'd be pretty sweet. Thanks again!

**Reply by JanDeDobbeleer** [reply]
Aaaaaah OK, missed that part. There's indeed nu such thing. In this case you can actually set that character regardless, it will work in any shell.

**Reply by regice202** [reply]
Well that's what it does in the standard unicorn theme, but like I said it shows one of those unknown character boxes is the standard PS windows. ![WeirdBox](https://i.imgur.com/HNK0ZVR.png) It only displays properly in Windows Terminal PS sessions.. So yeah, I think a template is the only chance to fix it.

**Reply by JanDeDobbeleer** [reply]
Allow me to deliver the advanced coloring features and I'm sure we can solve this.

**Reply by regice202** [reply]
Sounds good. Do you have a link for that issue so I can follow progress?

**Reply by JanDeDobbeleer** [reply]
It's a local branch. Allow me to push it tomorrow.

**Reply by regice202** [reply]
Perfect, thanks man!

**Reply by JanDeDobbeleer** [reply]
@regice202 executionTime now has the template functionality, meaning you can leverage the `template` property (and set pre and postfix to `""`) to add that braille icon when in windows terminal.

**Reply by regice202** [reply]
@JanDeDobbeleer Thank you so much for your time and hard work! Although I am a bit confused on how to execute this properly... The docs don't seem to be fully updated/clear on how to accomplish this. What would an example config look like for an <code>executiontime</code> block with this new change?...

**Reply by JanDeDobbeleer** [reply]
@regice202 different segment, but the same in regards top functionality is git. There's a [test](https://github.com/JanDeDobbeleer/oh-my-posh/blob/d422325f98206b239e6680d0c3ea75da7085e5ec/src/segment_git_test.go#L724) that illustrates how the template properties van be used.

**Reply by regice202** [reply]
@JanDeDobbeleer This is my current config for ``executiontime``, still haven't figured it out.. :/ 
```
{
  "type": "executiontime",
  "style": "diamond",
  "foreground": "#ffffff",
  "background": "#83769c",
  "leading_diamond": "",
  "trailing_diamond": "",
    "properties": {
      "always_enabled": true,
      "template": " \ufbab{{if .Env.IsWindowsTerminal eq \"1\"}}\u2800{{.FormattedMs.String}}\u2800{{end}}"
  }
}
```

**Reply by JanDeDobbeleer** [reply]
@regice202 `if eq .Env.IsWindowsTerminal \"1\"` -> functions always before the variables
